### PR TITLE
Move `fastANI` to a separate `requirements`

### LINKED ISSUE #377: `fastani` is not installed under Python 3.9
#### Summary:

The third-party dependency `fastani` cannot be installed with `conda` or `mamba` under Python 3.9.

#### Description:

`fastani` is an item in `requirements-thirdparty.txt` but is not installable under Python 3.9:

```bash
$ python --version
Python 3.9.9
$ conda install fastani
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
[...]                                                                                                                                                                                       UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - fastani -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0']

Your python: python=3.9

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.
```

Consequently, the `pytest -v` test fails:

```bash
$ pytest -v
========================================== test session starts ==========================================
platform darwin -- Python 3.9.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /Users/lpritc/opt/anaconda3/envs/pyani_py39/bin/python
cachedir: .pytest_cache
rootdir: /Users/lpritc/Development/GitHub/pyani, configfile: pytest.ini
plugins: ordering-0.6, cov-3.0.0
collected 118 items
[...]
tests/test_subcmd_09_fastani.py::TestfastANISubcommand::test_fastani FAILED                       [ 97%]
[...]
=============================================== FAILURES ================================================
__________________________________ TestfastANISubcommand.test_fastani ___________________________________

self = <test_subcmd_09_fastani.TestfastANISubcommand testMethod=test_fastani>

    def test_fastani(self):
        """Test fastani run."""
        print(self.argsdict["fastani"])
>       subcommands.subcmd_fastani(self.argsdict["fastani"])

tests/test_subcmd_09_fastani.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pyani/scripts/subcommands/subcmd_fastani.py:289: in subcmd_fastani
    run_fastani_jobs(job, args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[...]
```

#### Reproducible Steps:

1. Attempt to install `pyani` under Python 3.9 in any standard way (e.g. `git` clone then `pip install -e .`, etc.) - `fastani` will not be installed.
2. Attempt to run `pytest -v` (if cloned) or a `pyani fastani` command

#### Current Output:

Failed tests under `pytest`, errors with `pyani fastani`.

#### Expected Output:

1. The `fastani` tests should be skipped in `pytest` if `fastani` is unavailable.
2. `pyani fastani` should report that `fastani` is not installed, and fail gracefully.

#### pyani Version:

v0.3.0 (commit b8722f0)

#### Python Version:

3.9

#### Operating System:

macOS

### PR DESCRIPTION
**Pull request by baileythegreen**
Sidesteps macOS/Python3.9 issues with the `fastANI` `conda` recipe.

Closes #377.

Please include a summary of the change and which issue is fixed. Please also include the motivation and context, and note the tests that apply to these changes.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update

## Action Checklist

- [x] Work on a single issue/concept (if there are multiple separate issues to address, please use a separate pull request for each)
- [ ] Fork the `pyani` repository under your own account (please [allow write access for repository maintainers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
- [x] Set up an appropriate development environment (please see `CONTRIBUTING.md`)
- [x] Create a new branch with a short, descriptive name
- [x] Work on this branch
  - [x] style guidelines have been followed
  - [x] new code is commented, especially in hard-to-understand areas
  - [x] corresponding changes to documentation have been made
  - [x] tests for the change have been added that demonstrate the fix or feature works
- [x] Test locally with `pytest -v` **non-passing code will not be merged**
- [x] Rebase against `origin/master`
- [x] Check changes with `flake8` and `black` before submission
- [x] Commit branch
- [x] Submit pull request, describing the content and intent of the pull request
- [ ] Request a code review
- [ ] Continue the discussion at [`Pull requests` section](https://github.com/widdowquinn/pyani/pulls) in the `pyani` repository

### CODE REVIEW

**Review by widdowquinn — COMMENTED** [review]
The changes to the docs are going in the right direction (addition of `fastani` in the early section, noting the issue with Python 3.9)

Removing `fastani` from the `requirements-thirdparty.txt` file is the right thing to do, but I don't think there's any advantage to having the extra `requirements-fastani.txt` file. A user can more simply and explicitly execute:

`conda install fastani`

than 

`conda install --file requirements-fastani.txt`

and that should be our advice. Please remove the `requirements-fastani.txt` file.

As far as how to compile `fastANI` goes, I think we should use the docs to point and link users to #377 in the repository, where improvements/additions can be shared for others to read, rather than making a single suggestion concrete in the text (and imposing further documentation maintenance on ourselves to keep up). Please reword that section.

**Comment by widdowquinn** [comment]
After discussion I recall our reasons for having the extra `requirements-fastani.txt` file. For the record, they are:

- stylistic consistency
- we can add other dependencies that are `fastani` specific to this file if we need them
- convenience for the `Makefile`
- arguably easier for users

File: README.md [hunk 1]
```diff
@@ -171,26 +171,68 @@ pip3 install pyani
...
 
 **The legacy BLAST executable available from NCBI will not run on macOS Big Sur.**
 
 If you wish to use `pyani blastall` or the `ANIblastall` method with the legacy `pyani` interface, you will require a locally-installed copy of this package. This is one of the packages linked in the `requirements-thirdparty.txt` file.
 
+### Note: Installing fastANI
+
+**`fastANI` installation (via `conda`) will fail for Python 3.9 on macOS.**
+
+If you are using Python ≥3.9 to run `pyani` on macOS, `fastANI` will fail to install due to errors in the `conda` recipe. Using an earlier version of Python may be the simplest solution. However, you can also install `fastANI` yourself—via `conda`—using a slightly altered version of the command produced from the `conda` recipe, as shown below.
```
Comment (widdowquinn) [inline README.md:199]: Maybe state the specific conda recipe version (earlier/later recipes may not have this problem)

File: README.md [hunk 2]
```diff
@@ -171,26 +171,68 @@ pip3 install pyani
...
+```bash
+c++ -O3 -DNDEBUG -std=c++11 -Isrc \
+-I ${ENV_DIR}/lib/include -mmacosx-version-min=10.7 \
+-stdlib=libc++ -Xpreprocessor -fopenmp -lomp \
+-DUSE_BOOST src/cgi/core_genome_identity.cpp -o fastANI \
+${ENV_DIR}/lib/libboost_math_c99.dylib \
+-lstdc++ -lz -lm
+```
+
+For a more technical overview of the issue, please see https://github.com/widdowquinn/pyani/issues/377.
```
Comment (widdowquinn) [inline README.md:230]: Maybe flag explicitly that other solutions might be found on this thread, too.

**Review by widdowquinn — COMMENTED** [review]
Just a couple of notes about minor extra detail in the docs.

File: README.md (hunk 1 above)
Comment (baileythegreen) [inline README.md:199]:
```suggestion
If you are using Python ≥3.9 to run `pyani` on macOS, `fastANI` will fail to install due to errors in the `conda` recipe (for build `h0a26cfa_0`). Using an earlier version of Python may be the simplest solution. However, you can also install `fastANI` yourself—via `conda`—using a slightly altered version of the command produced from the `conda` recipe, as shown below.
```

Comment (baileythegreen) [reply README.md:199]:
```suggestion
If you are using Python ≥3.9 to run `pyani` on macOS, `fastANI` will fail to install due to errors in the `conda` recipe (for build `h0a26cfa_0`). Using an earlier version of Python may be the simplest solution. However, you can also install `fastANI` yourself—via `conda`—using a slightly altered version of the command produced from the `conda` recipe, as shown below.
```

File: README.md (hunk 2 above)
Comment (baileythegreen) [reply README.md:230]:
```suggestion
For a more technical overview of the issue, and other solutions that may be shared there, please see https://github.com/widdowquinn/pyani/issues/377.
```

**Pull request merged**